src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/client/activitiesappointments/ActivitiesAppointmentsClient.kt

```diff
@@ -73,29 +73,25 @@ class ActivitiesAppointmentsClient(private val activitiesAppointmentsApiWebClien
     .block()
 
   /**
-   * Returns all matching appointment (types) for a prisoner, not just video link bookings.
+   * Returns all matching appointment (types) for a prisoner at a given location, not just video link bookings.
    */
-  fun getPrisonersAppointmentsAtLocations(prisonCode: String, prisonerNumber: String, onDate: LocalDate, vararg locationIds: Long) = if (locationIds.isNotEmpty()) {
-    log.info("A&A CLIENT: query params - prisonCode=$prisonCode, prisonerNumber=$prisonerNumber, onDate=$onDate, locationIds=${locationIds.toList()}")
-    getPrisonersAppointments(prisonCode, prisonerNumber, onDate)
-      .filter { locationIds.toList().contains(it.internalLocation?.id) }
-  } else {
-    emptyList()
+  fun getPrisonersAppointments(prisonCode: String, prisonerNumber: String, onDate: LocalDate, locationId: UUID) = run {
+    log.info("A&A CLIENT: query params - prisonCode=$prisonCode, prisonerNumber=$prisonerNumber, onDate=$onDate, locationId=$locationId")
+    activitiesAppointmentsApiWebClient.post()
+      .uri("/appointments/{prisonCode}/search", prisonCode)
+      .bodyValue(
+        AppointmentSearchRequest(
```

Comment:
So this request does also support category code (appointment type) we might be able to harness this as well.